internal/controller/stats_reporter.go

```diff
@@ -0,0 +1,144 @@
+package controller
+
+import (
+	"context"
+	"regexp"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	scope = "sigs.k8s.io/secrets-store-sync-controller"
+
+	// Validating Admission Policy names that are monitered by metrics
+	CreateUpdateVAP               = "secrets-store-sync-controller-create-update-policy"
+	CreateUpdateTokenDenyVAP      = "secrets-store-sync-controller-create-update-token-deny-policy"
+	UpdateLabelVAP                = "secrets-store-sync-controller-update-label-policy"
+	UpdateOwnersCheckOldObjectVAP = "secrets-store-sync-controller-update-owners-check-oldobject-policy"
+	ValidateTokenVAP              = "secrets-store-sync-controller-validate-token-policy"
+)
+
+var (
+	wasSuccessfulKey = attribute.Key("successful")
+	namespaceKey     = attribute.Key("namespace")
```

Comment:
Don't use metrics labels with unbounded value domain, this can easily lead to state explosion and thus DOSing of the prometheus registry (https://github.com/kubernetes/kubernetes/pull/102523)